sql/src/main/java/io/crate/execution/jobs/JobSetup.java

```diff
@@ -886,33 +924,33 @@ public Boolean visitNestedLoopPhase(NestedLoopPhase phase, Context context) {
 
         @Override
         public Boolean visitHashJoinPhase(HashJoinPhase phase, Context context) {
-            MergePhase leftMerge = phase.leftMergePhase();
-            MergePhase rightMerge = phase.rightMergePhase();
-            RamAccountingContext ramAccountingLeft = leftMerge == null ? null :
-                RamAccountingContext.forExecutionPhase(breaker(), leftMerge);
-            RamAccountingContext ramAccountingRight = rightMerge == null ? null :
-                RamAccountingContext.forExecutionPhase(breaker(), rightMerge);
-
-            RamAccountingContext ramAccountingContext = RamAccountingContext.forExecutionPhase(breaker(), phase);
-            RowConsumer lastConsumer = context.getRowConsumer(phase, Paging.PAGE_SIZE, ramAccountingContext);
-            MemoryManager memoryManager = memoryManagerFactory.getMemoryManager(ramAccountingContext);
+            CircuitBreaker breaker = breaker();
+            int ramAccountingBlockSizeInBytes = BlockBasedRamAccounting.calculateBlockSizeInBytes(
+                breaker.getLimit(),
+                1
+            );
+            var ramAccounting = ConcurrentRamAccounting.forCircuitBreaker(phase.label(), breaker);
+            var ramAccountingOfOperation = new BlockBasedRamAccounting(
+                ramAccounting::addBytes,
+                ramAccountingBlockSizeInBytes);
+            RowConsumer lastConsumer = context.getRowConsumer(phase, Paging.PAGE_SIZE, ramAccountingOfOperation);
+            var memoryManager = memoryManagerFactory.getMemoryManager(ramAccounting);
+            var memoryManagerLeft = memoryManagerFactory.getMemoryManager(ramAccounting);
```

Comment:
Same here regarding the MemoryManager instances. Sorry I had only pointed it out once as I wasn't sure if I'm missing something.

Reply:
ah thanks!